lib/repoman/license_checker.rb

```diff
@@ -2,28 +2,17 @@ module Repoman
   class LicenseChecker < Checker
 
     def run
-      check_content FileFetcher.fetch(license_url)
-    rescue OpenURI::HTTPError => e
-      if e.message =~ /404/
-        add_error "No LICENSE file"
-        result
-      else
-        raise e
-      end
+      has_file, content = RequiredFileChecker.new(repo, 'LICENSE', 'No LICENSE file').fetch
+      has_file ? check_content(content) : add_error(content)
```

Comment:
This feels a bit like it's masking the 404 in the name of allowing a neat two line thing here. It also means that the second returned variable from `fetch` is overloaded as both content and error message, which feels… unclean to me?